[Fui Jing Min] iP

Fui03 opened the pull request:
# Lama :sunglasses:

> “You may delay, but time will not.” – Benjamin Franklin ([Source](https://www.chanty.com/blog/productivity-quotes/))

## Lama notes

- **Text-Based**
- **Fast Respond**
- **Clear Guide** :bookmark_tabs:

## Feature
- [X] Todo List
- [X] Mark down complete task
- [X] Delete finished task
- [X] Find your task
- [x] GUI

## _How to use!_
1. [Download](https://github.com/Fui03/ip/releases/download/A-Jar/lama.jar) the file
2. Use your terminal and direct it to the file directory
3. Run
```terminal
java -jar lama.jar
```

## Alternative choice!
1. Clone the source code [here](https://github.com/Fui03/ip.git)
2. Find the lama class and run:
```java
public static void main(String[] args) {
        new Lama("./data/lama.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Fui03 committed:
Level 0, change class name and greetings

Fui03 committed:
Level 1, Echos

Fui03 committed:
Level 2, add list

Fui03 committed:
Level 3, Created new class Task and mark as done feature

Fui03 committed:
Level 4, TODOs, events, deadlines

Fui03 committed:
Test Case

Fui03 committed:
Level 5, Handle Errors, Update Test Case

Fui03 committed:
Level 6, Delete Feature, update test case

Fui03 committed:
Basic Feature to store data

Fui03 committed:
Modify the data saved by append the task added, format the style

Fui03 committed:
Merge branch 'branch-Level-7'
Merge level 7 feature into master

Fui03 committed:
Date and time for deadline and event

Fui03 committed:
Merge branch 'branch-Level-8'
Level 8 Date time feature

Implement the date time for both deadline and event.

Fui03 committed:
Task list added and ui file with welcome added

Fui03 committed:
Complete OOP

Add parser and command for a better OOP. Modify the storage and ui file
to fulfil the OOP rules.

Fui03 committed:
Modify the style of codes

Fui03 committed:
Modify parser to take in capital command

Fui03 committed:
Merge branch 'branch-A-MoreOOP'
Modify codes by following better OOP rules

Added command, parser, task list and ui file. Modify storage file.

* Command have 6 subclasses to run the task.
* Parser deals  with the command given by user.
* TaskList is an array list of task.
* Ui deals with user interface.
* Storage help to store and load the user list.

Fui03 committed:
Divide classes into packages

Fui03 committed:
Merge branch 'branch-A-Packages'
Divide classes into packages

Created command and task package.

Fui03 committed:
Merge branch 'add-gradle-support'
Add gradle support

Fui03 committed:
Edit the gradle build

Fui03 committed:
Merge branch 'branch-A-Gradle'
Changes the build of gradle

Fui03 committed:
Test cases for tasklist

Fui03 committed:
Create test case for ui and standardise the tasklist test case

Make all the task list test case name end with Test.

Fui03 committed:
Test cases for storage

Fui03 committed:
Test cases for parser and modify the parser

Fui03 committed:
Test cases for deadline

Fui03 committed:
Create test cases for todo and event, modify deadline test cases

Fui03 committed:
Test cases for task

Fui03 committed:
Test cases for addcommand and modify the storage test cases

Fui03 committed:
Test cases for delete command

Fui03 committed:
Test cases for list command and modify add, delete command codes

Fui03 committed:
Test case for exit command

Fui03 committed:
Test case for command, modify the exit command test case

Fui03 committed:
Test case for mark command

Fui03 committed:
Test case for unmark command and modify mark command test case

Fui03 committed:
Merge branch 'branch-A-JUnit'
Add JUnit test cases for all classes

The current situation is that all the public method lack comprehensive
testing situation need to change to ensure that the code works correctly
and interacts correctly with its depedencies.

Write JUnit test cases to test and ensure all the functions run
correctly and interacts with UI.

This approach is chosen to ensure the code works correctly as expected.

All the test cases pass.

Fui03 committed:
Java doc for task class

Fui03 committed:
Java doc for todo class

Fui03 committed:
Java doc for event class

Fui03 committed:
Java doc for deadline class and modify the event java doc

Fui03 committed:
Java doc for command class

Fui03 committed:
Java doc for both add and delete command class

Fui03 committed:
Java doc for exit command and fix the typo of delete command class

Fui03 committed:
Modify the java doc for exit command class

Fui03 committed:
Java doc for list command class

Fui03 committed:
Java doc for mark command class

Fui03 committed:
Java doc for unmark command class

Fui03 committed:
Java doc for storage class

Fui03 committed:
Standardise the java doc format

Fui03 committed:
Java doc for UI class

Fui03 committed:
Java doc for parser class

Fui03 committed:
Java doc for task list class

Fui03 committed:
Java doc for lama class

Fui03 committed:
Java doc for lama exception class

Fui03 committed:
Java doc for deadline test class

Fui03 committed:
Java doc for todo test class

Fui03 committed:
Java doc for event test class

Fui03 committed:
Java doc for task test class

Fui03 committed:
Java doc for add command test class

Fui03 committed:
Java doc for command test class

Fui03 committed:
Java doc for delete command test class

Fui03 committed:
Java doc for exit command test class

Fui03 committed:
Java doc for list command test class

Fui03 committed:
Java doc for mark and unmark command test class

Fui03 committed:
Java doc for parser test class

Fui03 committed:
Java doc for storage test class

Fui03 committed:
Java doc for task list test class

Fui03 committed:
Java doc for ui test class

Fui03 committed:
Standardise the java doc

Fui03 committed:
Standardise the codes

Fui03 committed:
Standardise the codes

Fui03 committed:
Find feature that allows user to search for the keyword

Fui03 committed:
Merge branch 'branch-A-JavaDoc'
Master branch lacks comprehensive documentation

Adding documentation comments improves code quality and readability.

Merge the branch containing java documentation into the master branch.

Merging the updates ensures that documentation is included in the main
codebase.

Java documentation align with the project coding standard.

Fui03 committed:
Merge the coding stadard branch into master

The main branch currently lack of formal coding standard.

Coding standards are issential for ensuring consistency across the
codebase.

Merge the branch containing the coding standards into the master
branch.

Merge this branch into master branch promotes the consistency and
quality across the project.

Fui03 committed:
Merge branch 'branch-Level-9'

Merge the find command to search for tasks by keyword

Currently the application doesn't have feature to find the task.

Users need a feature to filter out the tasks based on the keywords to
improve task management.

Implemnt a find feature that allows user to search for tasks by a
keyword.

The find feature is implemented as a new command and test with unit test
case to ensure it flows correctly.

The feature includes JUnit test cases to verify its functionality.

Fui03 committed:
Debug the missing import for FindCommand in Parser

The Parser class fails to recognize the newly implemented FindCommand,
leading to erros when users try to execute the find command.

Without importing the FindCommand, the Parser cannot correctly process
the find command, resulting in a non-functional feature.

Add the import statement for FindCommand in the Parser class, ensuring
that the command is correctly executed.

Add the import resolves the issue without altering the existing command
structure, ensure the command run correctly.

After import, the functionailty is tested to confirm that the find
command works as expected.

Fui03 committed:
Merge branch 'branch-A-CodingStandard'

Merge bracnch-A-CodingStandard to the master branch

SQ77 commented on src/main/java/lama/Lama.java:
```diff
@@ -0,0 +1,63 @@
+    private TaskList taskList;
+
+    /**
+     * Construct a Lama object with specified file path for storing data.
+     * Initialise the UI, storage and task list.
```
Should this be "constructs" and "initialises" instead to comply with the coding standard?

SQ77 commented on src/main/java/lama/TaskList.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Remove the task at the specified index in the task list.
+     * @param i Integer index of the task in the task list.
```
Should there be a blank line in between these two lines?

SQ77 commented on src/main/java/lama/Ui.java:
```diff
@@ -0,0 +1,168 @@
+     *
+     * @param taskList TaskList that contains the matched tasks.
+     */
+    public void showFindCommand(TaskList taskList) {
```
Would the param name "filteredList" be better here to indicate the method expects a filtered list that contains the matched tasks?

SQ77 commented on src/main/java/lama/command/AddCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+/**
+ * Represent a command to add a task to the task list.
+ * Subclass of command.
```
Is this line of comment really needed?

SQ77 commented on src/main/java/lama/command/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class FindCommand extends Command {
+
+    private String word;
```
Could this variable be named more descriptively, such as "keyword"?

SQ77 commented on src/main/java/lama/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+     *
+     * @return String representation of the task suitable to save in the file.
+     */
+    public abstract String toFile();
```
Is the name of this method misleading? It seems to suggest that the method converts something to a file.

SQ77 reviewed (commented):
Overall, your code is well-organised and easy to read. Just a few minor details to take note of such as this one: "In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... etc. (not Return or Returning etc.)".

Fui03 committed:
Enforce coding style with checkstyle

Currently, the codebase lacks consistent enforcemnt of coding style
guide, leading to potential readability and maintenance issues.

This commit integrates checkstyle to automatically detect and enforce
coding style violation.

Configure checkstyle with rules that align with this project's coding
standards. Run checkstyle on the existing codebase and fix identified
issues

This approach ensures consistency, improves readability and maintains
coding standards throughout the project.

Fix all the files with violation to coding standards.

Fui03 committed:
Merge branch 'branch-A-CheckStyle'
Merge checkstyle integration into master branch

Integrates CheckStyle into the project to enforce consistent coding
standard.This merge ensure all code in the master branch adheres to the
specified guidelines, improving the code readability and mainainability.

Fui03 committed:
Setup for JavaFX

Fui03 committed:
Construct basic GUI

Fui03 committed:
Modify the GUI by implementing the CSS files

Fui03 committed:
Wrap up the GUI files into one package and modify the CSS of main window

Fui03 committed:
Complete the feature of GUI

Fui03 committed:
Modify some of the javaDoc to adhere the coding standard

Fui03 committed:
Modify the parser to identify blankspace

Fui03 committed:
JavaDoc for Launcher class

Fui03 committed:
Merge branch 'branch-Level-10'
Merge GUI updates and JavaFx

The GUI integration with JavaFx is incomplete, leading to inconsistent
user behavior. JavaDoc annotation are incomplete across several classes,
cause documentation gaps. Additionally, the parser incorrectly handles
commands that end with a space, resulting in unexpected errors.

A fully integrated GUI is crucial for ensuring a smooth user experience.
Complete and accurate JavaDoc is essential for ongoing maintenance and
understanding codebase. Proper handling of trailing spaces in parser
inputs is necessary to prevent runtime exception and improve command
processing reliability.

Merge all recent GUI enhancements into master branch. Update and
complete JavaDoc annotations for better clarity and compliance. Modify
the parser to trim trailing spaces from commands before processing.

Merging GUI updates enhances feature coherence and stability. Updating
JavaDoc ensures compliance with coding standards and improves code
usability for future development. Triming spaces in the parser avoid
unnecessary errors and align with best practices for input violation.

Documentation has been reviewed to ensure most major methods and classes
are covered.

Fui03 committed:
Debug on exit handling for JavaFX

The application does not correctly close when the "bye" command is
issued in the JavaFX GUI.

The improper shutdown handling can leave the JavaFX runtime hanging,
leading to potential resource management issue and a poor user
experience.

Use Platfrom.exit() in the ExitCommand class to properly terminate the
JavaFX application.

Platform.exit() allows for a graceful shutdown of the JavaFX platform.

This changes provide a reliable and user-friendly way to exit the
applicaiton, avoiding abrupt terminations.

harithh07 commented on src/main/java/lama/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @return The size of task list.
+     */
+    public int size() {
```
Should this method be named getSize() instead?

harithh07 commented on src/main/java/lama/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * Mark the task as undone.
+     */
+    public void markAsUnDone() {
```
Should this be changed to markAsUndone since undone is one word?

harithh07 commented on src/main/java/lama/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+     * @param word The keyword to search for in the task description.
+     * @return TaskList contain tasks that matched the word given.
+     */
+    public TaskList find(String word) {
```
Maybe the parameter can be renamed to keyword to be consistent?

harithh07 commented on src/main/java/lama/command/AddCommand.java:
```diff
@@ -0,0 +1,47 @@
+        ui.showAddCommand(taskList);
+        storage.addTask(task);
+        String output = "Got it. I've added this task:\n"
+                + "  " + taskList.get(taskList.size() - 1) + "\n"
```
Should this whitespace be included in the String before this?

harithh07 commented on src/main/java/lama/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+     *
+     * @return false by default.However, if command given should exit the loop, it will return true.
+     */
+    public boolean isExit() {
```
Should this be renamed to follow the convention for a boolean method? Eg. shouldExit or isExited

harithh07 reviewed (commented):
LGTM! Just some minor nits to fix.

Fui03 committed:
Clean up storage for future debug on list command

The list command returns "Here is your list" even when the list is
empty.

The behaviour is incorrect, as no list should be displayed when it is
empty.

Clean up storage to prepare for debugging and  fixing this issue.

Proper storage cleanup is needed to accurately reflect list data and
resolve the false output.

Future debugging will focus on identifying the source of the incorrect
list ouput.

Fui03 committed:
Fix bug in list command output when list is empty

Previously, the list command would incorrectly return "Here is your
list:" even when there were no items.

This behaviour was misleading, as no list should be displayed when the
list is empty.

Update the output to return "There is nothing in your list!" when the
list is empty.

This ensures the output is accurate and reflects the correct state of
the list.

The bug is now fixed and will no longer show a misleading message when
the list is empty.

Fui03 committed:
Assertions are missing in several classes

Without proper assertions, crticial assumptions about input validity are
not enforced, which could lead to errors or unexpected behaviour,
especially when interacting with user input, storage and tasks.

Add assertions across classes to validate inputs and enforce
assumptions. Ensure inputs are not null and valid.

Assertions provide a low overhead way to validate critical conditions
during development and testing. They allow the program to fail fast and
early if unexpected situations arise, making debugging and issue
tracking more efficient.

JUnit tests are also updated to check for assertion failures using
assertionThrows for cases where invalid input is provided, ensuring the
code behaves as expected during development.

Fui03 committed:
Merge pull request #1 from Fui03/branch-A-Assertions

Assertions are missing in several classes

Fui03 committed:
Refactor Storage class to improve code readability and maintainability

The Storage class handles file operations, including loading saving and
appending tasks. The method is long and nested.

The date format and task status reduce code readabilty, making it harder
to maintain and extend in future. Magic strings can lead to mistakes and
incosistencies when handling tasks.

Introduce constants for task status and date format. Refactor task
parsing into sperate helper methods for each task type to improve
clarity and seperation of concerns.

Using constants improve readability and reduces the chance of errors
caused by magic strings. Seperating task parsing logic into individual
methods makes the code easier to follow, debug and extends in future.

These changes prepare the code for future enhancement while ensuring
that existing functionality is preserved and easier to manage.

Fui03 committed:
Refactor Parser class and debug Find command output

The Parser class contains duplicated logic for handling various
commands. Additionally, the FindCommand is not returning the correct
output when searching for tasks based on keywords.

The duplicated logic increase code complexity and reduces
maintainability, while the bug in the FindCommand affects the program's
functionality, leading to incorrect search results.

Refactor the Parser class to extract repeated logic into helper methods
for better readability and maintainability. Debug and fix the issue in
the FindCommand to ensure it returns the correct tasks when searching by
keywords.

Extracting repeated logic reduces duplication, simplifies the code
structure and makes future maintainence easier. Fixing the FindCommand
ensures the search functionality works as intended, improving overall
user experience.

The focus is on improving code quality.

Fui03 committed:
Refactor Lama and Ui classes to improve code structure

The Lama and Ui classes contain repetitive logic, have methods that are
doing more than one task, and are missing some seperation of concerns.
Additionally, exception handling in the Lama class is mixed with the
initialization logic, reducing readability and maintainability.

To improve code readability and maintainability, it is necessary to
reduce methode size, seperate concerns and handle exception in a
cleaner, more modular way. Repetitive code and mixed responsibilities
make it harder to manage in future changes or debug issues.

Refactor the Lama and Ui classes by:
* Extracting task list initialization into a seperate method in Lama
  class.
* Extracting the command processing loop into a helper method in Lama
  class.
* Reducing duplication in Ui by creating helper methods for printing
  bars and task lists.

Extracting logic into helper methods to improve readability by making
each method smaller and more focused.

The changes are localized to Lama and Ui classes, preserving existing
functionality while improving code structure and quality.

Fui03 committed:
Refactor Deadline class by extracting date formats into constants

The Deadline class contains hardcoded date format for displaying and
saving deadlines, which makes the code less maintainable.

Hardcoded date format make the code harded to modify and maintain.
Extracting them into constants improves readability and fexibility.

Extract the date formats into constants.

Using constants for date formats ensures that any future changes be made
in one place, improving maintainability.

Fui03 committed:
Refactor DeleteCommand, FindCommand and ListCommand classes

The DeleteCommand, FindCommand and ListCommand classes contain repeared
logic and could benefit from improved readability and seperation of
concerns.

Refactoring these commands will simplify the methods, improve error
handling and enhance readability by breaking down logic into smaller
helper methods.

Refactor classes by:
* Refactor DeleteCommand to extract output generation logic into helper
  method, improving readability and error handling.
* Refactor FindCommand to simplify the task-matching logic.
* Refactor ListCommand by creating a helper method for building task
  list output.

Breaking logic into helper methods making each method more focused.

These changes maintain the same functionality but improve the code
quality.

Fui03 committed:
Refactor Main class structure and error handling

The Main Class in the JavaFx application directly handles FXML loading
and scene setup in the start() method, which reduces the clarity and
maintainability. Exception handling is minimal, using only
e.printStackTrace() for IOException.

To improve readability, maintainability and error handling, the code
should seperate concerns by using helper methods and provide more
informative error message when issues arise.

Improve the class by extracting FXML loading and controller setup
logic into a helper method loadMainWindow().

These changes improve the seperation of concerns and make the Main class
easier to understand and maintain.

The core functionality remains unchanged, but refactoring improves code
clarity and maintainability.

Fui03 committed:
Refactor Ui class by extracting common logic

The Ui class contains some repetitive code for displaying tasks leading
to redudancy in mulitple methods.

Code duplication makes maintenance more difficult and increases the
chances of incosistentcies.

Extract the common logic for displaying task lists into the
showTaskList() method and add showTaskCount() to handle repeated task
count display logic.

By centralizing repetitive code in helper methods, the class becomes
more readable and maintainable.

The core functionality remains unchanged.

Fui03 committed:
Merge branch 'master' of https://github.com/Fui03/ip into branch-A-CodeQuality

Merge master to the branch-A-CodeQuality and resolve merge conflict

The current branch has been merged with the master branch, which
introduced the assertions and changes. A conflict occurred in the
data/lama.txt file, which stores the task data.

The conflict in data/lama.txt needs to be resolved to ensure proper data
storage and consistency between both branches. Without resolving this,
the application may not handle data properly.

Resolve the conflict in data/lama.txt by integrating the necessary
changes from both branches to ensure the data is stored correctly.

This ensures that the task data is managed correctly and both the
assertions from the master branch and changes from the current branch
are integrated without losing important functionality.

All conflicts have been resolved and the data file is now consistent
with both branches. Besides a checkstyle is runned and resolve the
coding style which not align with the requirement.

Fui03 committed:
Merge pull request #2 from Fui03/branch-A-CodeQuality

Merge the branch-A-CodeQuality into master

Fui03 committed:
Add Gradle CI workflow using GitHub Actions

The project does not currently have a Continuous Integration (CI) setup
to automatically build and test the codebase on each push or pull
request. This is essential to ensure code quality and prevent issues
from being introduced.

Implementing CI is crucial to automatically run tests and validate code
changes.

Setup the CI workflow where it will be trigger on push or pull request
to master branch.

Using GitHub Actions provides an integrated CI platform directly within
GitHub.

This setup ensures that the project is built and tested automatically
with every push or pull request.

Fui03 committed:
Update CI workflow to run on Windows platform only

Fui03 committed:
Add alias functionality to commands

The application currently suports a fixed commands that requires users
to type out full command each time. There is no built-in functionality
for customising or shortening commands with user-defined aliases.

Allowing users to create aliases will improve the user experience by
enabling quicker and more flexible command entry. This will especially
benefit users who frequently use long commands.

Introduce a feature to allow users to set custom aliases for existing
commands. Add the AliasManager to handle alias storage and retrieval.
Implement new commands to manage aliases. Update parser to resolve
aliases.

By using an AliasManager, we centralize alias handling, making it easier
to maintain. The introduction of seperate commands for alias management
ensures clear and isolated functionality. Stroign aliases allows
persistence across sessions ,improving usability.

Default aliases like 't' for 'todo' are provided to streamline common
operations. Users can override these default as needed.

Fui03 committed:
Add Junit tests, modify the naming and handle empty input

The application has alias functionality but lacks proper JUnit test
coverage to ensure its reliability. Additionally, there is no proper
handling for empty input.

Without JUnit tests, there is no automated way to verify that alias
commands behave correctly, potentially introducing bugs.

Modify the classes by:
* Added JUnit test cases for AliasCommand, DeleteAliasCommand and
  AliasManger to validate alias creation and deletion.
* Rename the deletealias to remove.
* Rename the command to aliases.
* Handle empty and invalid inputs.

Adding test cases ensure that alias-related feature are verified for
correctness. Renaming the command make it easier for user to understand
while input validation safeguards the system against potential misuse.

These improvements enhance the robustness, usability and maintainability
of the alias system.

Fui03 committed:
Refactor Parser to resolve command after splitting the input

The Parser class resolves the command using AliasManager before
splitting.

Resolving the command after splitting the input ensures that the command
is processed correctly and avoids unecessary repetition.

Refactor the code to split the input into words first and then resolve
command using AliasManager.

By splitting the input first, we gurantee that the AliasManager
processes the command correctly.

This change is a minor change focuses on improving the efficiency in the
command parsing logic.

Fui03 committed:
Merge branch 'branch-C-FriendlierSyntax'
Merge aliases into master

The aliases feature has been developed in a seperate branch and is ready
to be merged into the master branch. This feature allows users to define
custom aliases for commands, improving the user experience by making
command input more flexible.

To bring the alias functionality into the main application, we need to
merge the branch into master. This will make the new alias capabilities
available in the production codebase and ensure future development
builds upon the latest code.

Merge the branch into master branch, resolving any conflicts and
ensuring the feature integrates smoothly with the existing code.

Merging this feature ensures that the new functionality is properly
integrated and tested within the branch. Merging into master allows
lastes features to be included.

All alias-related feature and tests have been implemented and validated
in the branch. The merge ensure all aliases are fully functional.

Fui03 committed:
Align code with coding standards

The codebase does not currently follow the project's established coding
standards, leading to incosistencies in code style and readability.

Enforce consitent formatting by applying the project's coding standards
to all files.

This is done to improve readability, maintainability and to adhere to
best practices.

No functional changes have been made.

Fui03 committed:
Add title and icon to JavaFX application

The application currently launches with the default window title and no
custom icon which affects the overall user experience.

Set a custom title for the application window and add an icon to improve
visual presentation and enhance the application.

This change is done to make the app more identifiable and polished from
the user's perspective.

No other functionality is impacted by this update.

Fui03 committed:
Setup data for GUI screenshot

The current storage data is not suitable for data displaying.

This needs to change to ensure a reasonable data is being output for the
screenshot purpose in future.

Edit the data in the data/lama.txt to make the data more reasonable.

This is done to ensure the data is better for displaying purpose.

No changes to functionality.

Fui03 committed:
Add a screenshot of the application

Fui03 committed:
Add user guide for the project

Fui03 committed:
Remove unnecessary imported button

Fui03 committed:
Modify to handle leading spaces in commands

Fui03 committed:
Modify the user guide

Fui03 committed:
Remove the last description in README.md

Fui03 committed:
Modify the description of README.md
